bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/internal/discovery/InboxResource.java

```diff
@@ -100,7 +103,7 @@ public Response approve(@HeaderParam(HttpHeaders.ACCEPT_LANGUAGE) @ApiParam(valu
         try {
             thing = inbox.approve(thingUIDObject, notEmptyLabel);
         } catch (IllegalArgumentException e) {
-            return JSONResponse.createErrorResponse(Status.NOT_FOUND, "Thing not found in inbox");
+            logger.error("Thing {} unable to be approved: {}", thingUID, e.getLocalizedMessage());
```

Comment:
Hm, don't we need still a return of "404"?

IMHO you should keep add

```
return JSONResponse.createErrorResponse(Status.NOT_FOUND, "Thing unable to be approved");
```

after the log mesage.

Reply:
Hmm - strange. I thought this was still there, but I guess the revert button removed it. I'll add it back - it was not intentional to remove it.